tools/generators/legacy/src/Generator/ConsolidateTargets.swift

```diff
@@ -392,6 +391,15 @@ extension ConsolidatedTarget {
                     rhsTarget.buildSettingConditional
             }
             .map { $1 }
+
+        if let moduleName = aTarget.product.moduleName,
+           sortedTargets.filter({ $0.product.moduleName == moduleName }).count == 1
```

Comment:
It's not the same. I checked that the product name only appears once, and your suggestion checks that there is no target with a different name. Additionally, it could behave incorrectly if some of the targets have nil as the module name.

Reply:
We need loop through all the elements of the array to see if there are any duplicates.